src/client/client.cpp

```diff
@@ -735,6 +725,72 @@ bool Client::loadMedia(const std::string &data, const std::string &filename,
 	return false;
 }
 
+bool Client::extractZipFile(const char *filename, const std::string &destination)
+{
+	auto fs = m_rendering_engine->get_filesystem();
+
+	if (!fs->addFileArchive(filename, false, false, io::EFAT_ZIP)) {
+		return false;
+	}
+
+	sanity_check(fs->getFileArchiveCount() > 0);
+
+	/**********************************************************************/
+	/* WARNING this is not threadsafe!!                                   */
+	/**********************************************************************/
+	io::IFileArchive* opened_zip = fs->getFileArchive(fs->getFileArchiveCount() - 1);
+
+	const io::IFileList* files_in_zip = opened_zip->getFileList();
+
+	unsigned int number_of_files = files_in_zip->getFileCount();
+
+	for (unsigned int i=0; i < number_of_files; i++) {
```

Comment:
Could this infinite loop? I also suggest ++i instead of i++ here but that's a very minor performance point.

Reply:
it's very minor and only called when loading media on connection